lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -792,7 +798,7 @@ def accessible_ancestor_objects(user_or_attribute, operation, root_object, optio
       # Fetch all of the PEAs using the given UA or its descendants
       user_attribute_ids = user_or_attribute.descendants.pluck(:id) | [user_or_attribute.id]
       associations = PolicyElementAssociation.where(user_attribute_id: user_attribute_ids)
-      operation_set_ids = associations.pluck(:operation_set_id)
+      operation_set_ids = OperationSet.where(id: associations.select(:operation_set_id)).pluck(:id)
```

Comment:
So, if a record is soft-deleted, that soft deletion will currently _not_ affect related records in the `associations`, `assignments`, etc. 

This fixes the issue by explicitly trying to find the Operation Set that is implied by the existence of the association, instead of relying on the implicit existence from the association. 

Interestingly enough, ActiveRecord/Arel is actually intelligent enough to _not_ create another SQL call! Instead, it will create a subquery!

Check this out: 
```sql
[9] pry(#<PolicyMachineStorageAdapter::ActiveRecord>)> puts OperationSet.where(id: associations.select(:operation_set_id)).to_sql
SELECT "policy_elements".* FROM "policy_elements" WHERE "policy_elements"."type" IN ('PolicyMachineStorageAdapter::ActiveRecord::OperationSet') AND "policy_elements"."id" IN (SELECT "policy_element_associations"."operation_set_id" FROM "policy_element_associations" WHERE "policy_element_associations"."user_attribute_id" IN (2, 1) AND "policy_element_associations"."object_attribute_id" IN (5, 6))
=> nil
```

A prettier version of the output: 
```sql
SELECT "policy_elements".*
FROM "policy_elements"
WHERE "policy_elements"."type" IN ('PolicyMachineStorageAdapter::ActiveRecord::OperationSet')
AND "policy_elements"."id" IN (
  SELECT "policy_element_associations"."operation_set_id"
  FROM "policy_element_associations"
  WHERE "policy_element_associations"."user_attribute_id" IN (2, 1)
    AND "policy_element_associations"."object_attribute_id" IN (5, 6)
)


```